iDiff/README.md

```diff
@@ -0,0 +1,244 @@
+| History                   | -d 	 | --history  |
+| npm installed packages    | -n 	 | --node     |
+| pip installed packages    | -p 	 | --pip      |
+| apt-get installed packages| -a 	 | --apt      |
+
+
+
+
+## Other Flags
+
+To get a JSON version of the iDiff output add a `-j` or `--json` flag.
+
+```iDiff <img1> <img2> -j```
+
+To use the docker client instead of shelling out to your local docker daemon, add a `-e` or `--eng` flag.
+
+```iDiff <img1> <img2> -e```
+
+
+## Output Format
+
+### History Diff
+
+The history differ has the following json output structure:
+
+```
+type HistDiff struct {
+	Image1 string
+	Image2 string
+	Adds   []string
+	Dels   []string
+}
+```
+
+### File System Diff
+
+The files system differ has the following json output structure: 
+
+```
+type DirDiff struct {
+	Image1 string
+	Image2 string
+	Adds   []string
+	Dels   []string
+	Mods   []string
+}
+```
+
+### Package Diffs
+
+Package differs such as pip, apt, and node inspect the packages contained within the images provided.  All packages differs currently leverage the PackageInfo struct which contains the version and size for a given package instance.
+
+```
+type PackageInfo struct {
+	Version string
+	Size    string
+}
+```
+
+#### Single Version Diffs
+
+The single version differs (apt, pip) have the following json output structure:
+
+```
+type PackageDiff struct {
+	Image1    string
+	Packages1 map[string]PackageInfo
+	Image2    string
+	Packages2 map[string]PackageInfo
+	InfoDiff  []Info
+}
+```
+
+Image1 and Image2 are the image names.  Packages1 and Packages2 map package names to PackageInfo structs which contain the version and size of the package.  InfoDiff contains a list of Info structs, each of which contains the package name (which occurred in both images but had a difference in size or version), and the PackageInfo struct for each package instance. 
+
+#### Multi Version Diffs
+
+The multi version differs (node) support processing images which may have multiple versions of the same package.  Below is the json output structure:
+
+```
+type MultiVersionPackageDiff struct {
+	Image1    string
+	Packages1 map[string]map[string]PackageInfo
+	Image2    string
+	Packages2 map[string]map[string]PackageInfo
+	InfoDiff  []MultiVersionInfo
+}
+```
+
+Image1 and Image2 are the image names.  Packages1 and Packages2 map package name to path where the package was found to PackageInfo struct (version and size of that package instance).  InfoDiff here is exanded to allow for multiple versions to be associated with a single package.
+
+```
+type MultiVersionInfo struct {
+	Package string
+	Info1   []PackageInfo
+	Info2   []PackageInfo
+}
+```
+
+## Known issues
+
+To run iDiff on image IDs or URLs, docker must be installed.
+
+## Example Run
+
+// TODO: update when output format finalized
+
+
+```
+$ iDiff  gcr.io/google-appengine/python:latest gcr.io/google-appengine/python:2017-07-25-110644 -p -d -a -n
```

Comment:
I think the output format just changed with the most recent PR merges, can you rerun the tool and put the new output here?